lib/pbench/server/database/models/datasets.py

```diff
@@ -913,10 +885,9 @@ class Metadata(Database.Base):
     key = Column(String(255), unique=False, nullable=False, index=True)
     value = Column(JSON, unique=False, nullable=True)
     dataset_ref = Column(Integer, ForeignKey("datasets.id"), nullable=False)
-    user_ref = Column(Integer, ForeignKey("users.id"), nullable=True)
 
     dataset = relationship("Dataset", back_populates="metadatas", single_parent=True)
-    user = relationship("User", back_populates="dataset_metadata", single_parent=True)
+    user_id = Column(String(255), nullable=True)
```

Comment:
Only `user.` namespace `Dataset` is owned (directly) by a user and has a `user_id` value. Metadata is essentially an extension of the `Dataset` so that's the primary foreign key and most metadata doesn't need a direct `owner_id` and will have `None` here. (Just as the previous `user_ref` column was `nullable=True`.)

Reply:
> most metadata doesn't need a direct `owner_id` and will have `None` here.

OK, thanks.

So, `relationship()` produces a nullable column by default?  Okie...nevermind.

Reply:
> So, relationship() produces a nullable column by default? Okie...nevermind.

Not quite. `relationship` is just a guide to SQLAlchemy for generating `join` statements when doing queries, and how to clean up when records are deleted. The connection is based on the foreign key column. Before, this was

```
user_ref = Column(Integer, ForeignKey("users.id"), nullable=True)
user = relationship("User", back_populates="dataset_metadata", single_parent=True)
```

So the table connection itself, through the `user_ref` foreign key, is nullable, but *if* there's a foreign key established, the `relationship` expression tells SQLAlchemy how to manage it.

We no longer have a `relationship` because we no longer have a `ForeignKey` column to manage: but the associated owner UUID is still nullable.